module/zfs/zfs_znode.c

```diff
@@ -653,7 +655,12 @@ zfs_inode_update(znode_t *zp)
 	ip->i_blkbits = SPA_MINBLOCKSHIFT;
 	ip->i_blocks = i_blocks;
 
-	ZFS_TIME_DECODE(&ip->i_atime, atime);
+	/*
+	 * Only read atime if we are new inode (or rezget), otherwise i_atime
+	 * might be dirty.
```

Comment:
I'd suggest rewording this so it's clear that "new inode" really means "newly created inode read from disk".